cql3/cql_statement.hh

```diff
@@ -49,8 +49,10 @@ public:
     // CQL statement text
     seastar::sstring raw_cql_statement;
 
-    // True for statements that needs guard to be taken before the execution
-    bool needs_guard = false;
+    // Returns true for statements that needs guard to be taken before the execution
+    virtual bool needs_guard(query_processor& qp) const {
```

Comment:
I recall that there was a story behind this that involved a large discussion between @gleb-cloudius and @avikivity , but I don't recall the specifics.

Reply:
The argument was that we should not make the code generic prematurely so @avikivity requested to drop the generic part of the patch despite the argument that it will be needed for auth (Avi's point was that we do not know how auth code will look like, so why assume things). Now we need the code to be more generic.

Reply:
so I guess now it not premature :) closing convo